### PR TITLE
chore: migrate to bun

### PR DESCRIPTION
**Pull request by flagrede**
## Description

Closes: regen-network/rnd-dev-team#1776

- migration to `Bun`
- Remove Lerna usage and  use Bun to manage the monorepo

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] provided a link to the relevant issue or specification
- [ ] provided instructions on how to test
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by aaronc** [comment]
Seems like Netlify officially supports Bun now: https://docs.netlify.com/configure-builds/manage-dependencies/#bun. Would like to see this get merged and have a modernized workflow

**Comment by aaronc** [comment]
Also happy to pick this up @flagrede if you're busy with other stuff. Seems like it should be relatively straightforward

**Comment by flagrede** [comment]
@aaronc That's great, I'll try to move forward with this PR today or tomorrow at worst.

**Comment by flagrede** [comment]
Bun is now set up for all packages. The only limitation is that I could not get absolute import to work with `web-components` (mainly because of storybook) but this could be a follow-up issue.

File: web-www/components/organisms/MarkdownSection/MarkdowmSection.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { ReactNode } from 'react';
 import { Box } from '@mui/material';
-import { makeStyles } from '@mui/styles';
 import clsx from 'clsx';
+import { makeStyles } from 'tss-react/mui';
```
Comment (aaronc) [inline web-www/components/organisms/MarkdownSection/MarkdowmSection.tsx:4]: Why is this change here?

**Review by aaronc — APPROVED** [review]

File: web-storybook/.storybook/main.js [hunk 2]
```diff
@@ -72,13 +72,6 @@ module.exports = {
             '../../web-marketplace/src/**/*.svg',
           ),
         }),
-        tsconfigPaths({
-          root: path.resolve(__dirname, '.'),
-          projects: [
-            path.resolve(__dirname, '../../web-components/tsconfig.json'),
-            path.resolve(__dirname, '../../web-marketplace/tsconfig.json'),
-          ],
-        }),
```
Comment (aaronc) [inline web-storybook/.storybook/main.js:81]: Why don't we need this?

**Comment by aaronc** [comment]
> Bun is now set up for all packages. The only limitation is that I could not get absolute import to work with `web-components` (mainly because of storybook) but this could be a follow-up issue.

Found this regarding bun and storybook: https://github.com/oven-sh/bun/issues/3794#issuecomment-1750986154

File: web-www/components/organisms/MarkdownSection/MarkdowmSection.tsx (hunk 1 above)
Comment (flagrede) [reply web-www/components/organisms/MarkdownSection/MarkdowmSection.tsx:4]: `makeStyles` from `@mui/styles'` is deprecated and should not be used anymore. We migrated all these imports to `tss-react/mui`, this is a remaining import from this migration.

File: web-storybook/.storybook/main.js (hunk 2 above)
Comment (flagrede) [reply web-storybook/.storybook/main.js:81]: I could not get `tsconfigPaths` to work with the `bun` migration (it's needed for absolute path to work in storybook). I think it's more related to our configuration rather than a bun-specific issue but as I was saying [here](https://github.com/regen-network/regen-web/pull/2139#issuecomment-1798621727) this is not blocking and we can investigate it further in a follow-up issue.

**Comment by flagrede** [comment]
> > Bun is now set up for all packages. The only limitation is that I could not get absolute import to work with `web-components` (mainly because of storybook) but this could be a follow-up issue.
> 
> Found this regarding bun and storybook: [oven-sh/bun#3794 (comment)](https://github.com/oven-sh/bun/issues/3794#issuecomment-1750986154)

Will try to replace the command before merging to see if it works well.
